crepecake/src/main/java/com/google/cloud/tools/crepecake/registry/ManifestPuller.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.registry;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.api.client.http.HttpResponseException;
+import com.google.cloud.tools.crepecake.blob.Blobs;
+import com.google.cloud.tools.crepecake.http.Authorization;
+import com.google.cloud.tools.crepecake.http.Connection;
+import com.google.cloud.tools.crepecake.http.Request;
+import com.google.cloud.tools.crepecake.http.Response;
+import com.google.cloud.tools.crepecake.image.json.ManifestTemplate;
+import com.google.cloud.tools.crepecake.image.json.UnknownManifestFormatException;
+import com.google.cloud.tools.crepecake.image.json.V21ManifestTemplate;
+import com.google.cloud.tools.crepecake.image.json.V22ManifestTemplate;
+import com.google.cloud.tools.crepecake.json.JsonTemplateMapper;
+import com.google.cloud.tools.crepecake.registry.json.ErrorEntryTemplate;
+import com.google.cloud.tools.crepecake.registry.json.ErrorResponseTemplate;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import javax.annotation.Nullable;
+
+/** Registry interface for pulling an image's manifest. */
+public class ManifestPuller {
+
+  // TODO: This should be configurable.
+  private static final String PROTOCOL = "http";
```

Comment:
should we be pulling over https?

Reply:
As discussed, https://github.com/GoogleCloudPlatform/minikube-build-tools-for-java/issues/78 filed to address this.